---
title: "sabrina-analysis"
author: "sabrina-mi"
date: "2020-07-20"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Definitions

```{bash, eval=FALSE}
conda activate imlabtools
GWAS_TOOLS=/Users/sabrinami/Github/summary-gwas-imputation/src
METAXCAN=/Users/sabrinami/Github/MetaXcan/software
DATA=/Users/sabrinami/Desktop/data
OUTPUT=/Users/sabrinami/Github/Workshop-GWAS-Summary-Harmonization/output
```


## Harmonization

```{bash}
python $GWAS_TOOLS/gwas_parsing.py \
-gwas_file $DATA/gwas/clozuk_pgc2.meta.sumstats.txt.gz \
-liftover $DATA/liftover/hg19ToHg38.over.chain.gz \
-snp_reference_metadata $DATA/reference_panel_1000G/variant_metadata.txt.gz METADATA \
-output_column_map SNP variant_id
-output_column_map A2 non_effect_allele \
-output_column_map A1 effect_allele \
-output_column_map CHR chromosome \
--chromosome_format \
-output_column_map BP position \
-output_column_map Freq.A1 frequency \
-output_column_map P pvalue \
-output_column_map OR or \
-output_column_map SE standard_error \
--insert_value sample_size 35802 --insert_value n_cases 11260 \
-output_order variant_id panel_variant_id chromosome position effect_allele non_effect_allele frequency pvalue zscore effect_size standard_error sample_size n_cases \
-output $OUTPUT/harmonized_gwas/clozuk_pgc2.meta.sumstats.txt.gz
```


## Imputation 
```{bash}
python $GWAS_TOOLS/gwas_summary_imputation.py \
-by_region_file $DATA/eur_ld.bed.gz \
-gwas_file $OUTPUT/harmonized_gwas/clozuk_pgc2.meta.sumstats.txt.gz \
-parquet_genotype $DATA/reference_panel_1000G/chr1.variants.parquet \
-parquet_genotype_metadata $DATA/reference_panel_1000G/variant_metadata.parquet \
-window 100000 \
-parsimony 7 \
-chromosome 1 \
-regularization 0.1 \
-frequency_filter 0.01 \
-sub_batches 10 \
-sub_batch 0 \
--standardise_dosages \
-output $OUTPUT/summary_imputation/clozuk_pgc2.meta.sumstats.txt.gz_chr1_sb0_reg0.1_ff0.01_by_region.txt.gz
```

## Post-Processing
```{bash}
python $GWAS_TOOLS/gwas_summary_imputation_postprocess.py \
-gwas_file $OUTPUT/harmonized_gwas/clozuk_pgc2.meta.sumstats.txt.gz \
-folder $OUTPUT/summary_imputation \
-pattern clozuk_pgc2.meta.sumstats.* \
-parsimony 7 \
-output $OUTPUT/processed_summary_imputation/clozuk_pgc2.meta.sumstats.txt.gz
```

## S-PrediXcan
```{bash}
python $METAXCAN/SPrediXcan.py \
--gwas_file  $OUTPUT/processed_summary_imputation/imputed_clozuk_pgc2.meta.sumstats.txt.gz \
--snp_column panel_variant_id --effect_allele_column effect_allele --non_effect_allele_column non_effect_allele --zscore_column zscore \
--model_db_path $DATA/models/eqtl/mashr/mashr_Brain_Cortex.db \
--covariance $DATA/models/eqtl/mashr/mashr_Brain_Cortex.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--throw \
--output_file $OUTPUT/spredixcan/eqtl/clozuk_pgc2_mashr_Brain_Cortex.csv
```

## S-PrediXcan Results
```{r}
spredixcan_association_mashr_Brain_Cortex = load_spredixcan_association(glue::glue("{OUTPUT}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_mashr_Brain_Cortex.csv"), gencode_df)
dim(spredixcan_association_mashr_Brain_Cortex)
significant_genes_mashr_Brain_Cortex <- spredixcan_association_mashr_Brain_Cortex %>% filter(pvalue < 0.05/nrow(spredixcan_association_mashr_Brain_Cortex)) %>% arrange(pvalue)
```

```{r plot brain cortex}
spredixcan_association_mashr_Brain_Cortex %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_mashr_Brain_Cortex$pvalue)
```
